pkg/templates/notificationtemplates/notification_generator.go

```diff
@@ -1 +1,61 @@
 package notificationtemplates
+
+import (
+	"strings"
+
+	"github.com/pkg/errors"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("notification-templates")
+
+// NotificationTemplate contains the template subject and content
+type NotificationTemplate struct {
+	Subject string
+	Content string
+}
+
+// GetNotificationTemplates returns a notification subject and body or an error
+func GetNotificationTemplates(name string, assets Assets) (*NotificationTemplate, error) {
+	templates, err := loadTemplates(assets)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get notification templates")
+	}
+	template := templates[name]
+	return &template, nil
+}
+
+func loadTemplates(assets Assets) (map[string]NotificationTemplate, error) {
+	paths := assets.Names()
+	templates := make(map[string]NotificationTemplate)
+
+	for _, path := range paths {
+		content, err := assets.Asset(path)
+		if err != nil {
+			return nil, err
+		}
+		segments := strings.Split(path, "/")
+		if len(segments) != 2 {
+			return nil, errors.Wrapf(errors.New("path must contain directory and file"), "unable to load templates")
+		}
+		directoryName := segments[0]
+		filename := segments[1]
+
+		if directoryName == "" || filename == "" {
```

Comment:
I don't think this case may happen: can you get an empty value from the `strings.Split(path, "/")` call above?

Reply:
i thought it would be possible if say generation when wrong and no assets were created etc. but i'll remove it

Reply:
done as part of: 32f0c75bea7b78ec9fd48726758ade2ba1b1ab66